ring-core/src/ring/util/response.clj

```diff
@@ -166,21 +175,36 @@
        (integer? (:status resp))
        (map? (:headers resp))))
 
-(defn- connection-content-length [resp ^java.net.URLConnection conn]
-  (let [content-length (.getContentLength conn)]
-    (if (neg? content-length)
-      resp
-      (header resp "Content-Length" content-length))))
+(defmulti resource-info
+  "Returns information about the resource specified by url.
+
+  The return value is a map with optional values for:
+
+  * :content        - the content of the URL, suitable for use as the :body
+                      of a ring response
+  * :content-length - the length of the :content, nil if not available
```

Comment:
`* :content-length` should be changed to `:content-length` to be consistent with other docstrings in Ring.